packages/report-server/src/reportBuilder/transform/aliases/keyValueByFieldAliases.ts

```diff
@@ -3,37 +3,56 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { Row } from '../../types';
+import { TransformTable } from '../table';
+import { FieldValue } from '../../types';
+
+const keyValueByField = (table: TransformTable, field: string) => {
+  const fieldValues = table.getColumnValues(field) as string[];
+  const valueValues = table.getColumnValues('value');
+  const newColumnNames = Array.from(new Set(fieldValues));
+  const newColumns: Record<string, FieldValue[]> = Object.fromEntries(
+    newColumnNames.map(columnName => [
+      columnName,
```

Comment:
Do you think it would be simpler to use the logic `get Rows -> modify Rows -> build table from Rows`?

Reply:
Hmmm yeah probably would be, I'll switch it it work that way 👍